pkg/credentials/credentials.go

```diff
@@ -147,11 +147,18 @@ func ProviderCredentials(p kubeone.CloudProviderName) (map[string]string, error)
 	return nil, errors.New("no provider matched")
 }
 
-func parseCredentialVariables(envVars []ProviderEnvironmentVariable, validationFunc func(map[string]string) error) (map[string]string, error) {
+func parseCredentialVariables(envVars []ProviderEnvironmentVariable, secrets *kubeone.KubeOneSecrets, validationFunc func(map[string]string) error) (map[string]string, error) {
 	// Validate credentials using given validation function
 	creds := make(map[string]string)
 	for _, env := range envVars {
-		creds[env.Name] = strings.TrimSpace(os.Getenv(env.Name))
+		if secrets != nil {
```

Comment:
As mentioned in the PR description, this allows both secrets and environment variables.